src/parcsr_mv/par_csr_matvec.c

```diff
@@ -117,167 +116,137 @@ hypre_ParCSRMatrixMatvecOutOfPlace( HYPRE_Complex       alpha,
       comm_pkg = hypre_ParCSRMatrixCommPkg(A);
    }
 
+   /* Update send_map_starts, send_map_elmts, and recv_vec_starts when doing
+      sparse matrix/multivector product  */
+   hypre_ParCSRCommPkgUpdateVecStarts(comm_pkg, x);
+
+   /* send_map_elmts on device */
+   hypre_ParCSRCommPkgCopySendMapElmtsToDevice(comm_pkg);
+
+   num_recvs = hypre_ParCSRCommPkgNumRecvs(comm_pkg);
    num_sends = hypre_ParCSRCommPkgNumSends(comm_pkg);
-   hypre_assert( num_cols_offd == hypre_ParCSRCommPkgRecvVecStart(comm_pkg,
-                                                                  hypre_ParCSRCommPkgNumRecvs(comm_pkg)) );
+
+   hypre_assert( num_cols_offd * num_vectors ==
+                 hypre_ParCSRCommPkgRecvVecStart(comm_pkg, num_recvs) );
+   hypre_assert( hypre_ParCSRCommPkgRecvVecStart(comm_pkg, 0) == 0 );
    hypre_assert( hypre_ParCSRCommPkgSendMapStart(comm_pkg, 0) == 0 );
 
 #ifdef HYPRE_PROFILE
    hypre_profile_times[HYPRE_TIMER_ID_PACK_UNPACK] -= hypre_MPI_Wtime();
 #endif
 
-   HYPRE_Int use_persistent_comm = 0;
-
-#ifdef HYPRE_USING_PERSISTENT_COMM
-   use_persistent_comm = num_vectors == 1;
```

Comment:
can we `use_persistent_comm` for `num_vectors` > 1?

Reply:
Sorry for the confusion earlier, but I tested this combination and `HYPRE_USING_PERSISTENT_COMM` is working with `num_vectors > 1` in this branch.